app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -206,20 +257,39 @@ public void setSuggestedSparkDriverMemory(long suggestedSparkDriverMemory) {
       suggestExecutorMemoryCore();
       suggestMemoryFactor();
       suggestDriverMemory();
+      suggestNumExecutorParameters();
       suggestedParameters.put(SparkConfigurationConstants.SPARK_EXECUTOR_MEMORY_KEY, (double) suggestedExecutorMemory
           / FileUtils.ONE_MB);
       suggestedParameters.put(SparkConfigurationConstants.SPARK_EXECUTOR_CORES_KEY, suggestedCore.doubleValue());
+      suggestedParameters.put(SparkConfigurationConstants.SPARK_EXECUTOR_CORES_KEY, suggestedCore.doubleValue());
       if (suggestedMemoryFactor < UnifiedMemoryHeuristic.SPARK_MEMORY_FRACTION_THRESHOLD()) {
         suggestedMemoryFactor = UnifiedMemoryHeuristic.SPARK_MEMORY_FRACTION_THRESHOLD();
       }
       suggestedParameters.put(SparkConfigurationConstants.SPARK_MEMORY_FRACTION_KEY, suggestedMemoryFactor);
       suggestedParameters.put(SparkConfigurationConstants.SPARK_DRIVER_MEMORY_KEY, (double) suggestedDriverMemory
           / FileUtils.ONE_MB);
+      if (suggestedDynamicAllocationMinExecutors != null) {
+        suggestedParameters.put(SparkConfigurationConstants.SPARK_DYNAMIC_ALLOCATION_MIN_EXECUTORS,
+            (double) suggestedDynamicAllocationMinExecutors);
+      }
+      if (suggestedDynamicAllocationMaxExecutors != null) {
+        suggestedParameters.put(SparkConfigurationConstants.SPARK_DYNAMIC_ALLOCATION_MAX_EXECUTORS,
+            (double) suggestedDynamicAllocationMaxExecutors);
+      }
+      if (suggestedExecutorInstances != null) {
+        suggestedParameters.put(SparkConfigurationConstants.SPARK_EXECUTOR_INSTANCES_KEY,
+            (double) suggestedExecutorInstances);
+      }
+
```

Comment:
can you look into  spark.dynamicAllocation.executorAllocationRatio (instead of setting min and max executors , these property is actually to decrease the resource consumption but doesn't work if min and max executors set explicitly)

Reply:
As discussed offline we will look into these parameters in detail later.

Reply:
ok

Reply:
While this config would be good to consider, but this was added only in Spark 2.4
Once we move to 2.4 in LinkedIn, we should consider it. But be careful of necessary checks as not everyone would be using 2.4